### PR TITLE
Fix make dependency

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

**Comment by yamt** [comment]
while working on macOS build jobs, i saw failures which look like incomplete Make.dep. i think this patch fixes them.

File: tools/LibTargets.mk [hunk 1]
```diff
@@ -194,7 +194,11 @@ libs$(DELIM)libxx$(DELIM)libxx$(LIBEXT): pass1dep
 staging$(DELIM)libxx$(LIBEXT): libs$(DELIM)libxx$(DELIM)libxx$(LIBEXT)
 	$(Q) $(call INSTALL_LIB,$<,$@)
 
+ifeq ($(CONFIG_BUILD_PROTECTED),y)
```
Comment (patacongo) [inline tools/LibTargets.mk:197]: Has this been tried CONFIG_BUILD_KERNEL?  It is a similar build in many ways.

Comment (yamt) [reply tools/LibTargets.mk:197]: i haven't tested CONFIG_BUILD_KERNEL.
my understanding is this doesn't matter for CONFIG_BUILD_KERNEL as it doesn't have libapps either in USERLIBS or NUTTXLIBS.

**Comment by acassis** [comment]
Hi @xiaoxiang781216 could you please take a look on this PR?

**Comment by yamt** [comment]
@xiaoxiang781216 are you still interested in this patch? i'm getting tired to rebase this.

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 are you still interested in this patch? i'm getting tired to rebase this.

@liuguo09 is working on this now. he can take care this patch.

**Comment by yamt** [comment]
@xiaoxiang781216 @liuguo09 ok. let me know if/when you want me rebase this.

**Comment by liuguo09** [comment]
Thanks @yamt.  I'm now looking into it and I'll rebase it later.

**Comment by xiaoxiang781216** [comment]
Merge through https://github.com/apache/incubator-nuttx/pull/1074

**Pull request closed without merging**